test/controllers/resource_controller_test.rb

```diff
@@ -0,0 +1,276 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+SingleCov.covered!
+
+describe "ResourceController Integration" do
+  let(:described_class) { ResourceController }
+
+  def assert_rendered_resource
+    json.keys.must_equal [:project]
+  end
+
+  def json
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  let(:project) { projects(:test) }
+
+  as_a :admin do
```

Comment:
Do you need "as_a :user" or "as_not_a :admin" ?

Reply:
need admin login here so all urls work like deleting commands etc